tests/embind/embind.test.js

```diff
@@ -479,19 +479,67 @@ module({
             assert.equal(expected, cm.get_non_ascii_wstring());
         });
 
-        test("passing unicode string into C++", function() {
+        test("non-ascii u16strings", function() {
+            var expected = String.fromCharCode(10) +
+                String.fromCharCode(1234) +
+                String.fromCharCode(2345) +
+                String.fromCharCode(65535);
+            assert.equal(expected, cm.get_non_ascii_u16string());
+        });
+
+        test("non-ascii u32strings", function() {
+            var expected = String.fromCharCode(10) +
+                String.fromCharCode(1234) +
+                String.fromCharCode(2345) +
+                String.fromCharCode(55357) +
+                String.fromCharCode(56833) +
+                String.fromCharCode(55357) +
+                String.fromCharCode(56960);
+            assert.equal(expected, cm.get_non_ascii_u32string());
+        });
+
+        test("passing unicode (wide) string into C++", function() {
             var expected = String.fromCharCode(10) +
                 String.fromCharCode(1234) +
                 String.fromCharCode(2345) +
                 String.fromCharCode(65535);
             assert.equal(expected, cm.take_and_return_std_wstring(expected));
         });
 
+        test("passing unicode (utf-16) string into C++", function() {
+            var expected = String.fromCharCode(10) +
+                String.fromCharCode(1234) +
+                String.fromCharCode(2345) +
+                String.fromCharCode(65535);
+            assert.equal(expected, cm.take_and_return_std_u16string(expected));
+        });
+
+        test("passing unicode (utf-32) string into C++", function() {
+            var expected = String.fromCharCode(10) +
```

Comment:
Can we pull these expected strings out of these inner test functions to avoid duplicating these large literals? Something like
```js
var utf16TestString = String.fromCharCode(10) +
  String.fromCharCode(1234) +
  String.fromCharCode(2345) +
  String.fromCharCode(65535);
var utf32TestString = String.fromCharCode(10) +
  String.fromCharCode(1234) +
  String.fromCharCode(2345) +
  String.fromCharCode(55357) +
  String.fromCharCode(56833) +
  String.fromCharCode(55357) +
  String.fromCharCode(56960);

// ...

test("passing unicode (utf-16) string into C++", function() {
  assert.equal(utf16TestString, cm.take_and_return_std_u16string(utf16TestString));
});

// ...
```